src/language-js/comments.js

```diff
@@ -942,6 +955,56 @@ function handleTSMappedTypeComments(
   return false;
 }
 
+function hasQuestionInRange(text, start, end, comments, options) {
+  for (let i = start; i < end; ++i) {
+    if (text.charAt(i) === "?") {
+      let hasQuestion = true;
+      // Ignore "?" in comments
+      for (const comment of comments) {
```

Comment:
Can we use https://github.com/prettier/prettier/blob/02137069a75260401498e1229f66edf5a9016d20/src/common/util.js#L243?


BTW: we really should use `token`  for this kind of stuff #8529

Reply:
I'll consider it.

> BTW: we really should use token for this kind of stuff #8529

Using `token` sounds good!

Reply:
I fixed with using `getNextNonSpaceNonCommentCharacterIndexWithStartIndex` for now. We should replace it with using `token` in future.